app/src/main/java/com/quarkworks/apartmentgroceries/grocery/AddGroceryItemActivity.java

```diff
@@ -48,46 +84,66 @@ protected void onCreate(final Bundle savedInstanceState) {
-        addButton = (Button) findViewById(R.id.add_grocery_item_add_button_id);
+        photoGridView = (GridView) findViewById(R.id.add_grocery_grid_view_id);
 
         /**
          * Set view data
          */
-        titleTextView.setText(getString(R.string.title_activity_add_grocery_item));
+        titleTextView.setText(getString(R.string.cancel));
+        titleRightTextView.setText(getString(R.string.post));
+
+        photoList = new ArrayList<>();
+        Bitmap cameraIconBitmap = getCameraIconBitmap();
+        ByteArrayOutputStream stream = new ByteArrayOutputStream();
+        cameraIconBitmap.compress(Bitmap.CompressFormat.PNG, 100, stream);
+        byte[] sampledInputData = stream.toByteArray();
+        photoList.add(sampledInputData);
+        imageAdapter = new ImageAdapter(this, photoList);
+        photoGridView.setAdapter(imageAdapter);
 
         /**
          * Set view OnClickListener
          */
-        addButton.setOnClickListener(addGroceryButtonOnClick);
+        titleTextView.setOnClickListener(cancelOnClick);
+        titleRightTextView.setOnClickListener(postOnClick);
 
         setSupportActionBar(toolbar);
         getSupportActionBar().setDisplayShowTitleEnabled(false);
     }
 
-    private View.OnClickListener addGroceryButtonOnClick = new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
-                String groceryItemName = groceryItemNameEditText.getText().toString();
+    private View.OnClickListener cancelOnClick = new View.OnClickListener() {
+        @Override
+        public void onClick(View v) {
+            HomeActivity.newIntent(AddGroceryItemActivity.this);
+        }
+    };
 
-                if (!groceryItemName.isEmpty()) {
-                    SharedPreferences sharedPreferences = getApplication()
-                            .getSharedPreferences(getString(R.string.login_or_sign_up_session), 0);
-                    String groupId = sharedPreferences.getString(RUser.JsonKeys.GROUP_ID, null);
-                    String userId = sharedPreferences.getString(RUser.JsonKeys.USER_ID, null);
+    private View.OnClickListener postOnClick = new View.OnClickListener() {
+        @Override
+        public void onClick(View v) {
+            String groceryItemName = groceryItemNameEditText.getText().toString();
 
-                    RGroceryItem rGroceryItem = new RGroceryItem();
-                    rGroceryItem.setName(groceryItemName);
-                    rGroceryItem.setGroupId(groupId);
-                    rGroceryItem.setCreatedBy(userId);
+            if (!groceryItemName.isEmpty()) {
+                SharedPreferences sharedPreferences = getApplication()
+                        .getSharedPreferences(getString(R.string.login_or_sign_up_session), 0);
+                String groupId = sharedPreferences.getString(RUser.JsonKeys.GROUP_ID, null);
+                String userId = sharedPreferences.getString(RUser.JsonKeys.USER_ID, null);
```

Comment:
Handle the backend stuff as a part remote sync functions rather than in the UI level code.